default/python/AI/AIstate.py

```diff
@@ -74,7 +74,12 @@ def convert_to_version(state, version):
 
     if version == 0:
```

Comment:
May be make some code that find all function with special name like 'convert_1_2',  then detect current code state and current save state and call all functions one by one.   

Maybe move this  functions to separate module, so they will not be present in main module. And it will be easy to disable code check for them.